voidcrew/code/modules/overmap/dynamic_events.dm

```diff
@@ -209,25 +226,24 @@
   * Unloads the reserve, deletes the linked docking port, and moves to a random location if there's no client-having, alive mobs.
   */
 /obj/structure/overmap/dynamic/proc/unload_level()
-	if(preserve_level)
+	if(preserve_level || concerned || !mapzone)
 		return
 
-	if(mapzone)
-		if(length(mapzone.get_mind_mobs()))
-			return //Dont fuck over stranded people? tbh this shouldn't be called on this condition, instead of bandaiding it inside
-		if(SSovermap.generator_type == OVERMAP_GENERATOR_SOLAR)
-			forceMove(SSovermap.get_unused_overmap_square_in_radius())
-		else
-			forceMove(SSovermap.get_unused_overmap_square())
-		choose_level_type()
-		remove_mapzone()
+	if(length(mapzone.get_mind_mobs()))
+		return //Dont fuck over stranded people? tbh this shouldn't be called on this condition, instead of bandaiding it inside
 
-	if(reserve_dock)
-		qdel(reserve_dock, TRUE)
-		reserve_dock = null
-	if(reserve_dock_secondary)
-		qdel(reserve_dock_secondary, TRUE)
-		reserve_dock_secondary = null
+	concerned = TRUE //Prevent someone to act with this while it reloads
+
+	remove_docks()
```

Comment:
Before your change this is supposed to be called regardless of if `mapzone` is null or not. Is there a reason you're changing it around?

Reply:
If it has no mapzone it shouldn't have docks.

Reply:
If it has dock with no mapzone, something going wrong, but anyway on new level generation dock will be overwritten, and landing check for mapzone, and if there no mapzone it generate new mapzone with new docks.

Reply:
Call. I check. And it replace all turfs by new space turfs.

Reply:
And for dock in space it create new obj